modules/va_facilities/lib/va_facilities/param_validators.rb

```diff
@@ -34,8 +41,102 @@ def validate_type_and_services_known
       raise Common::Exceptions::InvalidFieldValue.new('services', unknown) unless unknown.empty?
     end
 
+    def validate_street_address
+      if params[:street_address]
+        raise Common::Exceptions::InvalidFieldValue.new('street_address', params[:street_address]) unless
+            params[:street_address].match?(/\d/)
+      end
+    end
+
+    def validate_drive_time
+      Integer(params[:drive_time]) if params[:drive_time]
+    rescue ArgumentError
+      raise Common::Exceptions::InvalidFieldValue.new('drive_time', params[:drive_time])
+    end
+
+    def validate_lat
+      Float(params[:lat]) if params[:lat]
+    rescue ArgumentError
+      raise Common::Exceptions::InvalidFieldValue.new('lat', params[:lat])
+    end
+
+    def validate_lng
+      Float(params[:lng]) if params[:lng]
+    rescue ArgumentError
+      raise Common::Exceptions::InvalidFieldValue.new('lng', params[:lng])
+    end
+
+    def validate_bbox
+      raise ArgumentError unless params[:bbox].nil? || params[:bbox]&.length == 4
+      params[:bbox]&.each { |x| Float(x) }
+    rescue ArgumentError
+      raise Common::Exceptions::InvalidFieldValue.new('bbox', params[:bbox])
+    end
+
+    def validate_a_param_exists(require_one_param)
+      lat_and_long = params.key?(:lat) && params.key?(:long)
+
+      if !lat_and_long && require_one_param.none? { |param| params.key? param }
+        require_one_param.each do |param|
```

Comment:
At this point (via the conditional) we've established that none of the required_params exist.
`unless params.key? param` (the line below) is redundant.
Thus, the exception is going to be raised on the first value of `require_one_param`.
So, this whole method can be reduced to:
```
def validate_a_param_exists(require_one_param)
  lat_and_long = params.key?(:lat) && params.key?(:long)
  if !lat_and_long && require_one_param.none? { |param| params.key? param }
    raise Common::Exceptions::ParameterMissing.new(require_one_param.first.to_s, detail: MISSING_FACILITIES_PARAMS_ERR)
  end
end
```
Also, I don't think there is a spec testing this failure path.  Can we get a spec testing the path where the error is raised?
```
it 'fails validation when none of the required params are present' do
      @dummy_class.params = { blah: '75075' }
      expect{@dummy_class.validate_a_param_exists(REQUIRE_ONE_PARAM)}.to raise_error
    end
```